src/components/activity-client-interface.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2018 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnalyticsContext} from '../proto/api_messages';
+import {ActivityIframePort} from 'web-activities/activity-ports';
+import {ActivityMode, ActivityResult} from 'web-activities/activities';
+
+/**
+ * @typedef {!Array<(string|boolean|number|null|!Array<(string|boolean|number|null)>)>} SerializedBuffer
+ */
+export class ActivityClientInterface{
```

Comment:
We need to see what's the best way to wrap around these APIs. Activities has two main constructs: ActivityPorts and ActivityPort. ActivityPort could represent an iframe or a popup. They are similar in how they start and end, but not in other details.

In SwG we have 3-4 notions (or maybe even more):
1. Popup
2. Dialog View iframe.
3. Toast iframe.
4. Service iframe.

Intuitively I feel like we'd be better of if we followed the initial API when wrapping. We could even call it `SwgActivityPorts` and `SwgActivityPort`. But let's discuss in more detail.